Changes:

* User {{item}} instead of "/dev/{{item}}" to allow usage of
directories.

* Modified prepare disk and activate disk to allow directories

* Modified the ceph.conf template to set the object namespace len
and object name len variables (required by directories.

Patch-Author: Jorge Niedbalski <jorge.niedbalski@linaro.org>

--
diff --git a/ansible/roles/ceph/tasks/osd.yml b/ansible/roles/ceph/tasks/osd.yml
index 0a99979..3ef322c 100644
--- a/ansible/roles/ceph/tasks/osd.yml
+++ b/ansible/roles/ceph/tasks/osd.yml
@@ -2,28 +2,22 @@
   file: src=/etc/ceph/ceph.client.bootstrap-osd.keyring state=link force=yes
         dest=/var/lib/ceph/bootstrap-osd/ceph.keyring
 
-- name: Check if disk /dev/{{item}} is prepared
-  shell: ceph-disk list | grep "ceph data" | grep /dev/{{item}}
+- name: Check if disk {{item}} is prepared
+  shell: ceph-disk list | grep "ceph data" | grep {{item}}
   ignore_errors: yes
   changed_when: false
   register: ceph_disk_list
 
-- name: Deactivate disk /dev/{{item}}
+- name: Deactivate disk {{item}}
   when: ceph_disk_list is defined and (ceph_disk_list.rc == 0) and ceph_force_prepare
   changed_when: false
-  command: ceph-disk deactivate --mark-out /dev/{{item}}1
+  command: ceph-disk deactivate --mark-out {{item}}
 
-- name: Check if disk /dev/{{item}} is mounted
-  shell: grep /dev/{{item}} /proc/mounts
-  ignore_errors: yes
-  changed_when: false
-  register: grep_proc_mounts
-
-- name: Prepare disk /dev/{{item}}
-  when: ((ceph_disk_list is defined and ceph_disk_list.rc != 0) or ceph_force_prepare) and (grep_proc_mounts is defined and grep_proc_mounts.rc != 0)
-  command: ceph-disk prepare --zap-disk --cluster ceph --cluster-uuid {{ceph_fsid}} /dev/{{item}}
+- name: Prepare disk {{item}}
+  when: ((ceph_disk_list is defined and ceph_disk_list.rc != 0) or ceph_force_prepare)
+  shell: mkdir -p {{item}} && chown ceph:ceph {{item}} && ceph-disk prepare --zap-disk --cluster ceph --cluster-uuid {{ceph_fsid}} {{item}}
   register: ceph_disk
 
-- name: Activate disk /dev/{{item}}
+- name: Activate disk {{item}}
   when: ceph_disk is defined and ceph_disk.changed
-  command: ceph-disk activate /dev/{{item}}1
+  command: ceph-disk activate {{item}}
diff --git a/ansible/roles/ceph/templates/ceph.conf b/ansible/roles/ceph/templates/ceph.conf
index bb0dcaf..554d1c2 100644
--- a/ansible/roles/ceph/templates/ceph.conf
+++ b/ansible/roles/ceph/templates/ceph.conf
@@ -12,8 +12,8 @@ auth_client_required = cephx
 public addr = {{ceph_public_addr}}
 
 # needed for raw directories rather than disks
-#osd max object name len = 256
-#osd max object namespace len = 64
+osd max object name len = 256
+osd max object namespace len = 64
 
 [client.images]
 keyring = /etc/ceph/ceph.client.images.keyring
